anta/device.py

```diff
@@ -96,7 +100,19 @@ async def collect_commands(self, commands: List[AntaCommand]) -> None:
         Args:
             commands: the commands to collect
         """
-        await asyncio.gather(*(self.collect(command=command) for command in commands))
+
+        async def collect_command(command: AntaCommand) -> None:
```

Comment:
I would implement `collect()` in `AntaDevice` with the caching logic. This logic should call a private method `_collect()` that would be an abstract method implemented in `AntaDevice` subclasses.